### PR TITLE
MLE-11570 | Node integration to provide session details

### PR DESCRIPTION
**Pull request by akshatlambdatest**
Summary: Node integration to provide session details

JIRA: https://lambdatest.atlassian.net/browse/MLE-11570

### CODE REVIEW

File: lib/automation_client.js [hunk 1]
```diff
@@ -557,40 +526,145 @@ AutomationApiClient.prototype.fetchTunnels = function(fnCallback) {
...
+        console.error("Error fetching listing of builds --> ", error);
+        return {
+            success: false,
+            error: "Unable to fetch build listing",
+            msg: "",
+            data: {},
+        };
+    }
+
+    if (!!!builds.data || _.isEmpty(builds.data)) {
```
Comment (Shivanshu-lambdatest) [inline lib/automation_client.js]: Redundant !! , Can we replace !!! by ! ?

File: lib/automation_client.js [hunk 2]
```diff
@@ -557,40 +526,145 @@ AutomationApiClient.prototype.fetchTunnels = function(fnCallback) {
...
+        return {
+            success: false,
+            error: "Build/Session params are required",
+            msg: "",
+            data: {},
+        };
+    }
+
+    // check for mandatory parameters
+    if (!!!params.buildName) {
```
Comment (Shivanshu-lambdatest) [inline lib/automation_client.js]: Redundant !! , Can we replace !!! by ! ?

File: lib/automation_client.js [hunk 3]
```diff
@@ -557,40 +526,145 @@ AutomationApiClient.prototype.fetchTunnels = function(fnCallback) {
...
+    // structure of response
+    // const data = {
+    //     success,
+    //     error,
+    //     msg
+    //     data,
+    // };
+
+    // fetching top builds sorted by descending order of their build ids
+    if (!!!params) {
```
Comment (Shivanshu-lambdatest) [inline lib/automation_client.js]: Redundant !! , Can we replace !!! by ! ?

Comment (akshatlambdatest) [reply lib/automation_client.js]: Done

File: lib/automation_client.js (hunk 2 above)
Comment (akshatlambdatest) [reply lib/automation_client.js]: Done

File: lib/automation_client.js (hunk 1 above)
Comment (akshatlambdatest) [reply lib/automation_client.js]: Done

**Pull request merged**